src/toaster.ts

```diff
@@ -0,0 +1,20 @@
+import { Result } from '@ucanto/interface'
+import { toast } from 'react-hot-toast'
+
+export const ucantoast = async (promise: Promise<Result>, options: any) => {
+  return toast.promise(new Promise(async (resolve, reject) => {
+    promise.then((result) => {
+      if (result.ok) {
+        resolve(result.ok)
+      } else {
+        console.error("toaster got a Ucanto error: ", result.error)
+        resolve(result.error)
+      }
+    })
+
+    promise.catch((err: unknown) => {
+      console.error("toaster caught an error: ", err)
+      reject(err)
+    })
+  }), options)
```

Comment:
I find this a little hard to follow...why not just chain:

```suggestion
  return toast.promise(
    promise
      .then((result) => {
        if (result.error) console.error("toaster got a Ucanto error: ", result.error)
        return result.error ?? result.ok
      }).catch((err: unknown) => {
        console.error("toaster caught an error: ", err)
        throw err
      })
  ), options)
```

...or with async/await:

```suggestion
  return toast.promise((async () => {
    try {
      const result = await promise
      if (result.error) console.error("toaster got a Ucanto error: ", result.error)
      return result.error ?? result.ok
    } catch (err) {
      console.error("toaster got a Ucanto error: ", result.error)
      throw err
    }
  })(), options)
```

BUUUT, shouldn't we throw when we get a `result.error` so the toast displays it as an error?

Reply:
oh yea - typo in my code, that should have been `reject(result.error)` - trying to avoid throwing `result.error` was basically the reason I formulated it the way I did - I don't have strong feelings about which way we write this, happy to go with the async/await version modified to `throw result.error`!

Reply:
ok done!